src/runtime/security/client_negotiation.cpp

```diff
@@ -16,15 +16,38 @@
 // under the License.
 
 #include "client_negotiation.h"
+#include "negotiation_utils.h"
+
+#include <dsn/dist/fmt_logging.h>
 
 namespace dsn {
 namespace security {
 
-client_negotiation::client_negotiation(rpc_session *session) : negotiation(session) {}
+client_negotiation::client_negotiation(rpc_session *session) : negotiation(session)
+{
+    _name = fmt::format("C_NEGO_L({})=>R({})",
+                        dsn_primary_address().to_string(),
+                        _session->remote_address().to_string());
+}
+
+void client_negotiation::start()
+{
+    ddebug_f("{}: start negotiation", _name);
+    list_mechanisms();
```

Comment:
Why name `list_mechanisms()`, I see the follow logic is `send`

Reply:
No, It's main purpose is to list mechanisms which are supported by itself ,  send is only one of it's work. He did change it's status also